xc/xenops_server_xen.ml

```diff
@@ -2574,6 +2580,25 @@ module Actions = struct
 					| _ -> ()
 			) (DB.read vm)
 
+	let maybe_update_pv_drivers_detected ~xc ~xs domid path =
+		let vm = get_uuid ~xc domid |> Uuidm.to_string in
+		Opt.iter
+			(function { VmExtra.persistent; non_persistent } ->
+				if not non_persistent.VmExtra.pv_drivers_detected then begin
```

Comment:
Indeed. I think that's reasonable as uninstalling the drivers should be followed by a reboot and is a minor use-case. So if the PV drivers have been seen once then they are assumed to remain for the life of the domain.